core/src/main/scala/kafka/server/metadata/ClientQuotaMetadataManager.scala

```diff
@@ -147,13 +148,13 @@ class ClientQuotaMetadataManager(private[metadata] val quotaManagers: QuotaManag
     // Convert entity into Options with sanitized values for QuotaManagers
     val (sanitizedUser, sanitizedClientId) = quotaEntity match {
       case UserEntity(user) => (Some(Sanitizer.sanitize(user)), None)
-      case DefaultUserEntity => (Some(ZooKeeperInternals.DEFAULT_STRING), None)
+      case DefaultUserEntity => (Some(ClientQuotaManager.DefaultString), None)
```

Comment:
We should use Option[BaseUserEntity] instead of a string. As mentioned previously (https://github.com/apache/kafka/pull/18641#discussion_r1922770484), there's no need for a specific string to represent the "default" value.